### DISCUSSION TITLE
Home Assistant integration (finally!)

### DISCUSSION THREAD

**Discussion by WesWesthaver** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#682] [General]
I started using the Lay-Z-Spa WiFi remote late in the season last year. It worked and I was happy to have it. Now that this new season is upon us I wanted to try to integrate the Lay-Z-Spa WiFi remote into my Home Assistant system. I typically use the spa in the evenings between 6pm and midnight. So I set up a few automations to:

1) Relay the current temperature (from weather.forecast_home temperature) to the Lay-Z-Spa WiFi remote's ambient temperature.

2) Schedule the spa to be ready for use by 6pm by setting the Lay-Z-Spa WiFi remote's "set ready time."

3) Shut the spa heater OFF at midnight.

The MQTT feature is great for integrating the spa into Home Assistant. Nice work!

So far the the Lay-Z-Spa WiFi remote has done a pretty good job of figuring out what time to start heating the spa to have it ready for use at 6pm. At midnight the heater shuts down for the night and just after midnight I "set ready time" again for 6pm to start the process over again.

What kinds of automations have you done with your spa?

**Comment by brettus90** [comment]
Can you advise how you are relaying the ambient temp too the module from HA 

this would be useful for me i have the rest worked out :)

**Reply by WesWesthaver** [reply]
Sure. I created an automation. Here's a screen-shot of the YAML:

[Screen-shot](http://www.westhaver.com/Screenshot_2024-03-19_12-20-54_Home_Assistant_YAML.png)

Here's the text (copy/paste) version:

```
alias: Hot Tub - Set Ambient Temperature
description: Sends ambient temperature from weather service to hot tub controller.
trigger:
  - platform: state
    entity_id:
      - weather.forecast_home
    attribute: temperature
condition:
  - condition: not
    conditions:
      - condition: state
        entity_id: weather.forecast_home
        state: unknown
      - condition: state
        entity_id: weather.forecast_home
        state: unavailable
action:
  - service: mqtt.publish
    metadata: {}
    data:
      topic: layzspa/command
      payload_template: |-
        {
          "CMD":14, 
          "VALUE":{{state_attr('weather.forecast_home','temperature')|round}}, 
          "XTIME":0, 
          "INTERVAL":0
        }
mode: single
```

Note: the "CMD":14, sets the temperature as a Fahrenheit value. Change this to "CMD":15, for Centigrade.

**Comment by visualapproach** [comment]
> So far the the Lay-Z-Spa WiFi remote has done a pretty good job of figuring out what time to start heating the spa to have it ready for use at 6pm. At midnight the heater shuts down for the night and just after midnight I "set ready time" again for 6pm to start the process over again.

Glad to hear, it took more effort than expected to implement it, so I'm happy it is of use.

> What kinds of automations have you done with your spa?

Well, I tied some automations to physical button presses on the spa display, like extending a sunroof/awning above the pool when I pressed the timer button (which I disabled from doing anything in the config)

**Comment by XXXDirk** [comment]
Hello WesWesthaver,

how do you set the value "set ready time" with MQTT in HA?

I have not found a corresponding entry in the MQTT settings?

Thank you very much.

**Reply by WesWesthaver** [reply]
```alias: Hot Tub - Set Daily Ready Time
sequence:
  - service: mqtt.publish
    metadata: {}
    data:
      topic: layzspa/command
      payload_template: |-
        {%- 
          set ready_time = as_timestamp
          (
            (now().year  | string) + '-' +
            (now().month | string) + '-' +
            (now().day   | string) + ' ' +
            '18:00:00'
          )
        %}
        {
          "CMD":20,
          "VALUE":1,
          "XTIME":{{ ready_time }},
          "INTERVAL":0
        }
mode: single
icon: mdi:clock-time-twelve-outline
```

This script has to be run sometime after midnight and sets the spa ready time to 6pm that day.

**Comment by JacekHA** [comment]
Will he explain how to integrate the monitor with Home Assistant step by step?
